assignments/this.js

```diff
@@ -13,14 +13,60 @@
 
 // code example for Window Binding
 
+window._ = {};
+
 // Principle 2
 
 // code example for Implicit Binding
 
+function foo() {
+    console.log(this.a);
+}
+
+const obj2 = {
+    a: 10,
```

Comment:
please avoid small declarations and keys like this, it can cause a lot of issues for you in the future when you or others have to look at your code and figure out what you're trying to achieve. This is a bad habit to put into practice, remember to practice like you play and you will perform better when you get into a group setting.